siswrap/wrapper_services.py

```diff
@@ -83,30 +83,71 @@ class Wrapper(object):
         QualityControl at the moment).
 
         Args:
-            runfolder: the name of the runfolder to use (not full path)
-            configuration_svc: the ConfigurationService for our config lookups
-            logger: logger object for printouts
+            params: Dict of parameters to the wrapper. Must contain the name of
+                    the runfolder to use (not full path). Can contain a YAML
+                    object containing the Sisyphus config to use, and a XML
+                    object cotaining the QC config to use.
+            configuration_svc: The ConfigurationService for our config lookups
+            logger: Logger object for printouts
 
-        FIXME: Raises
+        Raises:
+            OSError: If the given runfolder doesn't exist.
     """
 
     QC_TYPE = "qc"
     REPORT_TYPE = "report"
 
-    def __init__(self, runfolder, configuration_svc, logger=None):
+    def __init__(self, params, configuration_svc, logger=None):
+        self.conf_svc = configuration_svc
+        self.logger = logger or logging.getLogger(__name__)
+
         conf = configuration_svc.get_app_config()
-        runpath = conf["runfolder_root"] + runfolder
+        runpath = conf["runfolder_root"] + "/" + params["runfolder"]
 
         if not os.path.isdir(runpath):
             raise OSError("No runfolder {0} exists.".format(runpath))
 
         self.info = ProcessInfo(runpath)
-        self.conf_svc = configuration_svc
-        self.logger = logger or logging.getLogger(__name__)
+
+        if "sisyphus_config" in params:
+            path = runpath + "/sisyphus.yml"
+            self.write_new_config_file(path, params["sisyphus_config"])
+
 
     def __get_attr__(self, attr):
         return getattr(self.info, attr)
 
+    @staticmethod
+    def write_new_config_file(path, content):
+        """ Writes new config file (especially used for Sisyphus YAML and QC XML).
+            If the file already exists a backup copy will be created.
+
+            Args:
+                - path: The path to the config file that should be written.
+                - content: The content of the new config file.
+        """
+        try:
+            import shutil
```

Comment:
I also have a tendency to want to put imports in as small a scope as possible, but the as you say the PEP states otherwise: https://www.python.org/dev/peps/pep-0008/#imports so I think we should move them.